lib/team_statistics.rb

```diff
@@ -0,0 +1,200 @@
+require_relative "game_data"
+require_relative "team_data"
+require_relative "game_team_data"
+require_relative "league_statistics"
+require "csv"
+
+class TeamStatistics < LeagueStatistics
+
+  def initialize
+    @team_info_by_id = Hash.new
+    @by_team_id_game_objects = []
+    @total_games_by_season = Hash.new{ |hash, key| hash[key] = 0 }
+    @total_wins_by_season = Hash.new{ |hash, key| hash[key] = 0 }
+    @win_percentage_by_season = Hash.new
+    @games_won_by_team_id = Hash.new{ |hash, key| hash[key] = 0 }
+    @goals_by_game = []
+    @team_name_by_id = Hash.new{}
+    @games_played_by_opponent = Hash.new{ |hash, key| hash[key] = 0 }
+    @games_won_by_opponent = Hash.new{ |hash, key| hash[key] = 0 }
+    @win_ratio_by_opponent = Hash.new
+  end
+
+  def all_teams
+    TeamData.create_objects
+  end
+
+  def all_games
+    GameData.create_objects
+  end
+
+  def all_game_teams
+    GameTeamData.create_objects
+  end
+
+  def team_info(passed_id)
+    all_teams.each do |team|
+      if passed_id == team.team_id.to_s # Remove .to_s when spec harness info updates
+        @team_info_by_id[:team_id] = team.team_id
+        @team_info_by_id[:franchise_id] = team.franchise_id
+        @team_info_by_id[:team_name] = team.team_name
+        @team_info_by_id[:abbreviation] = team.abbreviation
+        @team_info_by_id[:link] = team.link
+      end
+    end
+    @team_info_by_id
+  end
+
+  def collect_game_objects_by_team_id(passed_id)
+    all_games.each do |game_object|
+      if game_object.home_team_id.to_s == passed_id
+        @by_team_id_game_objects << game_object
+      elsif game_object.away_team_id.to_s == passed_id
+        @by_team_id_game_objects << game_object
+      end
+    end
+  end
+
+  def total_games_by_season_by_team_id
+    @by_team_id_game_objects.each do |game|
+      @total_games_by_season[game.season] += 1
+    end
+  end
+
+  def total_wins_by_season_by_team_id(passed_id)
+    @by_team_id_game_objects.each do |game|
+      if passed_id == game.away_team_id.to_s && game.away_goals > game.home_goals
+        @total_wins_by_season[game.season] += 1
+      elsif passed_id == game.home_team_id.to_s && game.home_goals > game.away_goals
+        @total_wins_by_season[game.season] += 1
+      elsif passed_id == game.away_team_id.to_s && game.away_goals < game.home_goals
+        @total_wins_by_season[game.season] += 0
+      elsif passed_id == game.home_team_id.to_s && game.home_goals < game.away_goals
+        @total_wins_by_season[game.season] += 0
```

Comment:
Let's not forget to delete all of the 'to_s' when moving forward and merging with Logan's code

Reply:
Great remind! Thanks :)